torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -274,12 +274,12 @@ struct ConstantPropagator {
 
       i++; // increment bc we didn't remove current index
     }
-    // an output was removed
-    return initial_outputs != true_block->outputs().size();
+    made_change_ |= initial_outputs != true_block->outputs().size();
```

Comment:
I think it is safer to track whether `replaceAndRemoveIfOutput` (e.g., make it return bool) made changes to the graph to set `made_change_`. In the future if `replaceAndRemoveIfOutput` may alter the graph without changing the number of nodes, the code would still be correct.

Reply:
i dont know how realistic a scenario that removeExtraIfOutputs is going to add new outputs.. but i can make the change

Reply:
i dont know how realistic a scenario that removeExtraIfOutputs is going to add new outputs.. but i can make the change